ibc/types.go

```diff
@@ -42,8 +42,8 @@ type ChainConfig struct {
 	PreGenesis func(ChainConfig) error
 	// When provided, genesis file contents will be altered before sharing for genesis.
 	ModifyGenesis func(ChainConfig, []byte) ([]byte, error)
-	// Modify genesis-amounts
-	ModifyGenesisAmounts func() (sdk.Coin, sdk.Coin)
+	// Modify genesis-amounts for the validator at the given index
+	ModifyGenesisAmounts func(int) (sdk.Coin, sdk.Coin)
```

Comment:
Should modify/add an e2e test to use this. Where 2 validators are used and the 2nd (index 1) has a different amount.

This is extremely used for Proof of Authority testing as well

Reply:
done; i didn't end up pulling StakingQueryDelegation just because there's no easy way (that I know) to get a node's validator address anyway. Instead i just called `query staking validators` directly and parsed that.